satellite/attribution/db_test.go

```diff
@@ -55,3 +98,181 @@ func TestDB(t *testing.T) {
+				end:     time.Date(now.Year(), now.Month(), now.Day()+1, 0, 0, 0, 0, now.Location()),
+				padding: 2,
+			},
+			{
+				name:       "partnerID, new projectID, alpha",
+				partnerID:  partnerID,
+				projectID:  newUUID(),
+				bucketName: alphaBucket,
+
+				remoteSize: remoteSize / 2,
+				inlineSize: inlineSize / 2,
+				egressSize: egressSize / 2,
+
+				start:   time.Date(now.Year(), now.Month(), now.Day(), 0, 0, 0, 0, now.Location()),
+				end:     time.Date(now.Year(), now.Month(), now.Day()+1, 0, 0, 0, 0, now.Location()),
+				padding: 2,
+			},
+			{
+				name:       "new partnerID, projectID, beta",
+				partnerID:  newUUID(),
+				projectID:  projectID,
+				bucketName: betaBucket,
+
+				remoteSize: remoteSize / 3,
+				inlineSize: inlineSize / 3,
+				egressSize: egressSize / 3,
+
+				start:   time.Date(now.Year(), now.Month(), now.Day(), 0, 0, 0, 0, now.Location()),
+				end:     time.Date(now.Year(), now.Month(), now.Day()+1, 0, 0, 0, 0, now.Location()),
+				padding: 2,
+			},
+			{
+				name:       "partnerID, new projectID, beta",
+				partnerID:  partnerID,
+				projectID:  newUUID(),
+				bucketName: betaBucket,
+
+				remoteSize: remoteSize / 4,
+				inlineSize: inlineSize / 4,
+				egressSize: egressSize / 4,
+
+				start:   time.Date(now.Year(), now.Month(), now.Day(), 0, 0, 0, 0, now.Location()),
+				end:     time.Date(now.Year(), now.Month(), now.Day()+1, 0, 0, 0, 0, now.Location()),
+				padding: 2,
+			}}
+
+		for _, td := range testData {
+			td := td
+			td.init()
+			info := attribution.Info{td.projectID, td.bucketName, td.partnerID, time.Time{}}
+			_, err := db.Attribution().Insert(ctx, &info)
+			require.NoError(t, err)
+
+			for i := 0; i < td.hoursOfData; i++ {
+				createData(ctx, t, db, &td)
+			}
+
+			verifyData(ctx, t, db.Attribution(), &td)
+		}
+	})
+}
+
+func verifyData(ctx *testcontext.Context, t *testing.T, attributionDB attribution.DB, testData *AttributionTestData) {
+	results, err := attributionDB.QueryAttribution(ctx, testData.partnerID, testData.start, testData.end)
+	require.NoError(t, err)
+	require.NotEqual(t, 0, len(results), "Results must not be empty.")
+	count := 0
+	for _, r := range results {
+		projectID, _ := bytesToUUID(r.ProjectID)
+		// The query returns results by partnerID, so we need to filter out by projectID
+		if projectID != testData.projectID {
+			continue
+		}
+		count++
+
+		assert.Equal(t, testData.partnerID[:], r.PartnerID, testData.name)
+		assert.Equal(t, testData.projectID[:], r.ProjectID, testData.name)
+		assert.Equal(t, testData.bucketName, r.BucketName, testData.name)
+		assert.Equal(t, float64(testData.expectedRemoteBytes/testData.hours), r.RemoteBytesPerHour, testData.name)
+		assert.Equal(t, float64(testData.expectedInlineBytes/testData.hours), r.InlineBytesPerHour, testData.name)
+		assert.Equal(t, testData.expectedEgress, r.EgressData, testData.name)
+	}
+	require.NotEqual(t, 0, len(results), "Results were returned, but did not match the projectID.")
```

Comment:
shouldn't here be `count` or something else than `len(results)`?

Reply:
Updated